### ISSUE TITLE
Invalid SPIR-V for bvec3 constructor

### ISSUE THREAD

**Issue by Qining** [KhronosGroup/glslang#177]
The SPIR-V of the following glsl code may have invalid constructor for bvec3.
The **bvec3(v_out0)** results into an object with type 'vector of three ints', while the expected type is 'vector of three bools'.

Then in the following OpStore operation `bvec3 out0 = <constructed object>`, the pointer operand type and object operand type does not match each other. 

```
#version 310 es
precision highp float;
precision highp int;

layout(location = 0) out mediump vec4 dEQP_FragColor;

bool isOk (bvec3 a, bvec3 b) { return (a == b); }

layout(location = 0) flat in ivec3 v_out0;
layout(binding = 0, std140) uniform Reference
{
        bvec3 out0;
} ref;

void main()
{
        bvec3 out0 = bvec3(v_out0);
        bool RES = isOk(out0, ref.out0);
dEQP_FragColor = vec4(RES, RES, RES, 1.0);
}

```

**Comment by Qining** [comment]
Following is the full SPIR-V code generated:

**%42 = OpLoad %28 %41
               OpStore %39 %42**
is where the mismatch happens.

**%8 = OpTypePointer Function %7**
%8 have %7 as its pointee type, while %42's type is %28.

And %7, %28 also have different element types.

```
; SPIR-V
; Version: 1.0
; Generator: Khronos Glslang Reference Front End; 1
; Bound: 57
; Schema: 0
               OpCapability Shader
          %1 = OpExtInstImport "GLSL.std.450"
               OpMemoryModel Logical GLSL450
               OpEntryPoint Fragment %4 "main" %24 %47
               OpExecutionMode %4 OriginLowerLeft
               OpSource ESSL 310
               OpSourceExtension "GL_GOOGLE_cpp_style_line_directive"
               OpSourceExtension "GL_GOOGLE_include_directive"
               OpName %4 "main"
               OpName %12 "isOk(vb3;vb3;"
               OpName %10 "a"
               OpName %11 "b"
               OpName %20 "out0"
               OpName %24 "v_out0"
               OpName %32 "RES"
               OpName %33 "Reference"
               OpMemberName %33 0 "out0"
               OpName %35 "ref"
               OpName %37 "param"
               OpName %39 "param"
               OpName %47 "dEQP_FragColor"
               OpDecorate %24 Flat
               OpDecorate %24 Location 0
               OpMemberDecorate %33 0 Offset 0
               OpDecorate %33 Block
               OpDecorate %35 Binding 0
               OpDecorate %47 RelaxedPrecision
               OpDecorate %47 Location 0
               OpDecorate %51 RelaxedPrecision
               OpDecorate %53 RelaxedPrecision
               OpDecorate %55 RelaxedPrecision
               OpDecorate %56 RelaxedPrecision
          %2 = OpTypeVoid
          %3 = OpTypeFunction %2
          %6 = OpTypeBool
          %7 = OpTypeVector %6 3
          %8 = OpTypePointer Function %7
          %9 = OpTypeFunction %6 %8 %8
         %21 = OpTypeInt 32 1
         %22 = OpTypeVector %21 3
         %23 = OpTypePointer Input %22
         %24 = OpVariable %23 Input
         %26 = OpTypeInt 32 0
         %27 = OpConstant %26 0
         %28 = OpTypeVector %26 3
         %29 = OpConstantComposite %28 %27 %27 %27
         %31 = OpTypePointer Function %6
         %33 = OpTypeStruct %28
         %34 = OpTypePointer Uniform %33
         %35 = OpVariable %34 Uniform
         %36 = OpConstant %21 0
         %40 = OpTypePointer Uniform %28
         %44 = OpTypeFloat 32
         %45 = OpTypeVector %44 4
         %46 = OpTypePointer Output %45
         %47 = OpVariable %46 Output
         %49 = OpConstant %44 0
         %50 = OpConstant %44 1
          %4 = OpFunction %2 None %3
          %5 = OpLabel
         %20 = OpVariable %8 Function
         %32 = OpVariable %31 Function
         %37 = OpVariable %8 Function
         %39 = OpVariable %8 Function
         %25 = OpLoad %22 %24
         %30 = OpINotEqual %7 %25 %29
               OpStore %20 %30
         %38 = OpLoad %7 %20
               OpStore %37 %38
         %41 = OpAccessChain %40 %35 %36
         %42 = OpLoad %28 %41
               OpStore %39 %42
         %43 = OpFunctionCall %6 %12 %37 %39
               OpStore %32 %43
         %48 = OpLoad %6 %32
         %51 = OpSelect %44 %48 %50 %49
         %52 = OpLoad %6 %32
         %53 = OpSelect %44 %52 %50 %49
         %54 = OpLoad %6 %32
         %55 = OpSelect %44 %54 %50 %49
         %56 = OpCompositeConstruct %45 %51 %53 %55 %50
               OpStore %47 %56
               OpReturn
               OpFunctionEnd
         %12 = OpFunction %6 None %9
         %10 = OpFunctionParameter %8
         %11 = OpFunctionParameter %8
         %13 = OpLabel
         %14 = OpLoad %7 %10
         %15 = OpLoad %7 %11
         %16 = OpLogicalEqual %7 %14 %15
         %17 = OpAll %6 %16
               OpReturnValue %17
               OpFunctionEnd

```

**Comment by johnkslang** [comment]
I think this is a duplicate of issue #170, which should be covering all the int <-> bool conversions.  Only scalar reads are working right now.

**Comment by Qining** [comment]
Close as this has already been fixed in 2725323bba789de1eb524f57ae5d2f0386d4f40a
